source/mir/ndslice/algorithm.d

```diff
@@ -1884,3 +1885,760 @@ size_t countImpl(alias fun, Slices...)(Slices slices)
     while(!slices[0].empty);
     return ret;
 }
+
+private template eachLowerTriangle(alias fun, size_t k = 1,
+                                                bool BeyondMainDiagonal = false)
```

Comment:
I think it would be better to have `eachLowerTriangle(alias fun,ptrdiff_t diagonalOffset)` if I am understanding this correctly.

Reply:
That's more-or-less how it originally was, but I thought the code was a little simpler when I had done it this way.